### ISSUE TITLE
upgrade async to 2.x+

### ISSUE THREAD

**Issue by barisusakli** [NodeBB/NodeBB#4938]
https://github.com/caolan/async/blob/master/CHANGELOG.md#breaking-changes

Breaks plugins if they do `module.parent.require('async');` and use any of the functions that have breaking changes.

**Comment by pitaj** [comment]
> Breaks plugins if they do module.parent.require('async');

WHY WOULD ANYBODY DO THAT ITS SO STUPID AGGGGGGGGGHHHHHH

**Comment by barisusakli** [comment]
```
filter, reject, some, every, and related functions now expect an error as the first callback argument, rather than just a simple boolean. Pass null as the first argument, or use fs.access instead of fs.exists. (#118, #774, #1028, #1041)
```
Plugins using the above needs to be fixed before 1.5.0

**Comment by julianlam** [comment]
:shipit:
